clippy_lints/src/methods/manual_contains.rs

```diff
@@ -0,0 +1,117 @@
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::peel_hir_pat_refs;
+use clippy_utils::source::snippet_with_applicability;
+use rustc_ast::UnOp;
+use rustc_errors::Applicability;
+use rustc_hir::def::Res;
+use rustc_hir::{BinOpKind, Body, Expr, ExprKind, HirId, QPath};
+use rustc_lint::LateContext;
+use rustc_middle::ty::{self};
+use rustc_span::Span;
+use rustc_span::source_map::Spanned;
+
+use super::MANUAL_CONTAINS;
+
+pub(super) fn check(cx: &LateContext<'_>, expr: &Expr<'_>, recv: &Expr<'_>, closure_arg: &Expr<'_>) {
+    let mut app = Applicability::MachineApplicable;
+
+    if !expr.span.from_expansion()
+        // check if `iter().any()` can be replaced with `contains()`
+        && let ExprKind::Closure(closure) = closure_arg.kind
+        && let Body{params: [param],value} = cx.tcx.hir().body(closure.body)
+        && let ExprKind::Binary(op, lhs, rhs) = value.kind
+        && let (peeled_ref_pat, _)=peel_hir_pat_refs(param.pat)
+        && let Some((snip,snip_expr))=can_replace_with_contains(cx, op, lhs, rhs, peeled_ref_pat.hir_id, &mut app)
+    {
+        let ref_type = cx.typeck_results().expr_ty_adjusted(recv);
+
+        match ref_type.kind() {
+            ty::Ref(_, inner_type, _) if inner_type.is_slice() => {
+                if let ty::Slice(slice_type) = inner_type.kind()
+                    && *slice_type == cx.typeck_results().expr_ty(snip_expr)
+                {
+                    span_lint_and_sugg(
+                        cx,
+                        MANUAL_CONTAINS,
+                        expr.span,
+                        "using `contains()` instead of `iter().any()` is more efficient",
+                        "try",
+                        format!(
+                            "{}.contains({})",
+                            snippet_with_applicability(cx, recv.span, "..", &mut app),
+                            snip
+                        ),
+                        app,
+                    );
+                }
+            },
+            _ => {},
+        }
+    }
+}
+
+enum EligibleArg {
+    IsClosureArg,
+    ContainsArg(String),
+}
+
+fn try_get_eligible_arg<'tcx>(
+    cx: &LateContext<'tcx>,
+    expr: &'tcx Expr<'tcx>,
+    closure_arg_id: HirId,
+    applicability: &mut Applicability,
+) -> Option<(EligibleArg, &'tcx Expr<'tcx>)> {
+    let mut get_snippet = |span: Span, needs_borrow: bool| {
+        let snippet = snippet_with_applicability(cx, span, "..", applicability).to_string();
+        if needs_borrow {
+            EligibleArg::ContainsArg(format!("&{snippet}"))
+        } else {
+            EligibleArg::ContainsArg(snippet)
+        }
+    };
```

Comment:
This won't work if `span` points to a binary expression and `needs_borrow` is `true`:

```rust
    let vec: Vec<u32> = vec![1, 2, 3, 4, 5, 6];
    let values = &vec[..];
    let _ = values.iter().any(|&v| v == 4 + 1);
```

will generate invalid fixed code.

You should use `Sugg` which takes care of adding the required parentheses if needed when referencing with `.addr()`:

```suggestion
    let mut get_snippet = |expr: &Expr<'_>, needs_borrow: bool| {
        let sugg = Sugg::hir_with_applicability(cx, expr, "_", applicability);
        EligibleArg::ContainsArg((if needs_borrow { sugg.addr() } else { sugg }).to_string())
    };
```

Can you add the above as a test also?